### PR TITLE
issue #2 - perspective transform

### PR DESCRIPTION
**Pull request by wojtke**

### CODE REVIEW

File: perspective_transform.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        List of four points arranged anticlockwise starting top-left corner
+    output_shape: tuple
+        Desired shape of transformed image (height, width)
+
+    Returns
+    -------
+    numpy.ndarray
+        Transformed image
+
+	"""
```
Comment (j-adamczyk) [inline perspective_transform.py:23]: Should be aligned (spaces), e.g.:
```
Parameters
    ----------
```
Both should be in the same line.

File: perspective_transform.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import numpy as np
+from skimage import data, transform
+import warnings
+
+def perspective_transform(img: np.ndarray, input_points: list, output_shape: tuple) -> np.ndarray:
```
Comment (j-adamczyk) [inline perspective_transform.py:5]: With many arguments it's more clear to write them in separate lines (see e.g. PyTorch or Scikit-learn):
```
def perspective_transform(
  img: np.ndarray, 
  input_points: list, 
  output_shape: tuple) \
  -> np.ndarray:
```

File: perspective_transform.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        Desired shape of transformed image (height, width)
+
+    Returns
+    -------
+    numpy.ndarray
+        Transformed image
+
+	"""
+
+	#transformation coordinates
```
Comment (j-adamczyk) [inline perspective_transform.py:25]: PEP8 - comments should have 1 space after `#` and before first letter

File: perspective_transform.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+	dst = np.array(input_points)
+	src = np.array([[0, 0], [0, output_shape[0]], [output_shape[1], output_shape[0]], [output_shape[1], 0]])
```
Comment (j-adamczyk) [inline perspective_transform.py:27]: Such "magic" with indices should be avoided, it's better to create variables, e.g. upper_left, upper_right etc.

File: perspective_transform.py [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+	#checking if input_points are correct 
+	if not dst.shape == (4,2):
+		warnings.warn(f"Shape of input_points expected to be (4,2), got {dst.shape} instead.\n") 
+	elif not (dst[0,1]<dst[1,1] and dst[1,0]<dst[2,0] and dst[2,1]>dst[3,1] and dst[3,0]>dst[0,0]):
+		warnings.warn("Suspicious arrangement of input_points, expected anticlockwise starting top-left corner.", )
```
Comment (j-adamczyk) [inline perspective_transform.py:33]: 1. Checks should be done at the very beginning of the function.
2. If the `dst.shape` is not `(4, 2)`, is it even possible to carry on with the function?
3. After `,` and before and after `<` or `>` there should be spaces, e.g. `dst[0, 1] < dst[1, 1]`. Also if you create variables as suggested in other comment above, you can avoid ugly index-based code.

File: perspective_transform.py [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+	#transformation
+	tform = transform.ProjectiveTransform()
+	tform.estimate(src, dst)
+	warped_img = transform.warp(img, tform, output_shape=output_shape)
+
+	return warped_img
+
+
+
```
Comment (j-adamczyk) [inline perspective_transform.py:43]: Only 1 empty line is expected at the end of file.

**Comment by wojtke** [comment]
Made changes you suggested. Code readability is not perfect, but greatly improved.

**Review by konrrad — APPROVED** [review]

**Review by j-adamczyk — APPROVED** [review]

**Pull request merged**